spring-pulsar/src/test/java/org/springframework/pulsar/core/DefaultSchemaResolverTests.java

```diff
@@ -189,11 +193,12 @@ static Stream<Arguments> primitiveMessageTypesProvider() {
 		@Test
 		void customMessageTypes() {
 			assertThat(resolver.getSchema(Foo.class, false)).isNull();
-			assertThat(resolver.getSchema(Foo.class, true)).isEqualTo(Schema.BYTES);
+			assertThat(resolver.getSchema(Foo.class, true).getSchemaInfo())
+					.isEqualTo(Schema.JSON(Foo.class).getSchemaInfo());
 			resolver.addCustomSchemaMapping(Foo.class, Schema.STRING);
 			assertThat(resolver.getSchema(Foo.class, false)).isEqualTo(Schema.STRING);
 			assertThat(resolver.getSchema(Bar.class, false)).isNull();
-			assertThat(resolver.getSchema(Bar.class, true)).isEqualTo(Schema.BYTES);
+			assertThatExceptionOfType(AvroTypeException.class).isThrownBy(() -> resolver.getSchema(Bar.class, true));
```

Comment:
But it's fine for `Bar`. Hmmm? I wonder how limiting this will be in practice? Many pojos are probably typed. 

I think since we are making this resolver smarter now, I would go w/ 
1. log a trace about this fact
2. default to bytes

I can envision us making the defaulting behavior easily configurable in the future (I mean, of course they can use a custom resolver)